src/workerd/io/worker.c++

```diff
@@ -2865,31 +2871,34 @@ void Worker::Actor::assertCanSetAlarm() {
   KJ_UNREACHABLE;
 }
 
-kj::Promise<void> Worker::Actor::makeAlarmTaskForPreview(kj::Date scheduledTime) {
+void Worker::Actor::Impl::HooksImpl::updateAlarmInMemory(kj::Maybe<kj::Date> newTime) {
+  if (newTime == nullptr) {
+    maybeAlarmPreviewTask = nullptr;
+    return;
+  }
+
+  auto scheduledTime = KJ_ASSERT_NONNULL(newTime);
+
   auto retry = kj::coCapture([this, originalTime = scheduledTime]()
       -> kj::Promise<void> {
     kj::Date scheduledTime = originalTime;
 
     for(auto i : kj::zeroTo(WorkerInterface::ALARM_RETRY_MAX_TRIES)) {
-      auto result = co_await impl->timerChannel.atTime(scheduledTime)
+      auto result = co_await timerChannel.atTime(scheduledTime)
           .then([this, originalTime]() {
-        return impl->loopback->getWorker(IoChannelFactory::SubrequestMetadata{})->runAlarm(originalTime);
+        return loopback->getWorker(IoChannelFactory::SubrequestMetadata{})->runAlarm(originalTime);
       });
 
       if (result.outcome != EventOutcome::OK && result.retry) {
         auto delay = (WorkerInterface::ALARM_RETRY_START_SECONDS << i++) * kj::SECONDS;
-        auto& timeContext = this->impl->timerChannel;
-        scheduledTime = timeContext.now() + delay;
+        scheduledTime = timerChannel.now() + delay;
       } else {
         break;
       }
     }
   });
 
-  auto task = retry().fork();
-
-  IoContext::current().addWaitUntil(task.addBranch());
-  return task.addBranch();
+  maybeAlarmPreviewTask = retry();
```

Comment:
I know the goal was to remove the IoContext dependency in the header, but just curious why it's removed it here too. I know we can recreate the actor as needed now, but since this is held by the `Worker::Actor::Impl::HooksImpl`, couldn't the `maybeAlarmPreviewTask` be dropped before it completes?